roles/pulp_user/tasks/main.yml

```diff
@@ -18,37 +20,52 @@
   when:
-    - user.json.count == 1
-    - user.item.state is defined
-    - user.item.state == "absent"
+    - users_get_result.results[user_index].json.count == 1
+    - pulp_users[user_index].state is defined
+    - pulp_users[user_index].state == "absent"
 
 - name: Set users to create fact
   set_fact:
-    create_users: "{{ create_users + [user.item] }}"
-  loop: "{{ users_get_result.results }}"
+    create_users: "{{ create_users + [pulp_users[user_index]] }}"
+  loop: "{{ users_get_result.results | map(attribute='item') }}"
   loop_control:
-    loop_var: "user"
-    label: "{{ user.item.username }}"
+    index_var: user_index
   when:
-   - user.json.count == 0
-   - user.item.state is not defined or user.item.state != "absent"
+    - users_get_result.results[user_index].json.count == 0
+    - pulp_users[user_index].state is not defined or pulp_users[user_index].state != "absent"
 
 - name: Set users to update fact
   set_fact:
-    update_users: "{{ update_users + [user.json.results[0] | combine(user.item) | combine({'existing_groups': user.json.results[0]['groups']}) ] }}"
-  loop: "{{ users_get_result.results }}"
+    update_users: >-
+      {{- update_users + [users_get_result.results[user_index].json.results[0]
+      | combine(pulp_users[user_index])
+      | combine({'existing_groups': users_get_result.results[user_index].json.results[0]['groups']})] -}}
+  loop: "{{ users_get_result.results | map(attribute='item') }}"
   loop_control:
-    loop_var: "user"
-    label: "{{ user.item.username }}"
+    index_var: user_index
   when:
-   - user.json.count == 1
-   - user.item.state is not defined or user.item.state != "absent"
+    - users_get_result.results[user_index].json.count == 1
+    - pulp_users[user_index].state is not defined or pulp_users[user_index].state != "absent"
+    - >-
+      users_get_result.results[user_index].json.results[0].username != pulp_users[user_index].username or
+      (pulp_users[user_index].first_name is defined and
+      users_get_result.results[user_index].json.results[0].first_name != pulp_users[user_index].first_name) or
+      (pulp_users[user_index].last_name is defined and
+      users_get_result.results[user_index].json.results[0].last_name != pulp_users[user_index].last_name) or
+      (pulp_users[user_index].email is defined and
+      users_get_result.results[user_index].json.results[0].email != pulp_users[user_index].email) or
+      (pulp_users[user_index].is_active is defined and
+      users_get_result.results[user_index].json.results[0].is_active != pulp_users[user_index].is_active) or
+      (pulp_users[user_index].is_staff is defined and
+      users_get_result.results[user_index].json.results[0].is_staff != pulp_users[user_index].is_staff) or
+      (pulp_users[user_index].groups is defined and
+      users_get_result.results[user_index].json.results[0].groups | map(attribute='name') | list != pulp_users[user_index].groups) or
+      pulp_users[user_index].password is defined
```

Comment:
This is a bit susceptible to breaking when new parameters are added, but hopefully that won't happen too often.

Reply:
Agreed. In general I'm not super happy with how it works but I think it's better to have it than not